neuralmonkey/tests/test_encoders_init.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3.5
+"""Test init methods of encoders."""
+
+import random
+import string
+import unittest
+
+from neuralmonkey.decoding_function import Attention, CoverageAttention
+from neuralmonkey.encoders.numpy_encoder import (VectorEncoder,
+                                                 PostCNNImageEncoder)
+from neuralmonkey.encoders.sentence_encoder import SentenceEncoder
+from neuralmonkey.tests.test_vocabulary import VOCABULARY
+
+SENTENCE_ENCODER = {
+    "max_input_len": ([1, 10, 100], [-1, 0, "ahoj", 3.14]),
+    "embedding_size": ([1, 10, 100], [-1, 0, "ahoj", 3.14]),
+    "rnn_size": ([1, 10, 100], [-1, 0, "ahoj", 3.14]),
+    "dropout_keep_prob": ([0.1, 1.0], [-1, 1.5, "huhu"]),
+    "attention_type": ([Attention, CoverageAttention, None], ["baf"]),
+    "use_noisy_activations": ([False], [0, "", "bflm"]),
+    "data_id": (["marmelade"], [0]),
+    "vocabulary": ([VOCABULARY], [0])
+}
+
+VECTOR_ENCODER = {
+    "data_id": (["marmelade"], [0]),
+    "dimension": ([1, 100], [0, -1, "hoj", 3.14]),
+    "output_shape": ([1, 100], [0, -1, "hoj", 3.14])
+}
+
+POST_CNN_IMAGE_ENCODER = {
+    "data_id": (["marmelade"], [0]),
+    "attention_type": ([Attention, CoverageAttention, None], ["baf"]),
+    "output_shape": ([1, 100], [0, -1, "hoj", 3.14]),
+    "input_shape": ([[1, 2, 3], [10, 20, 3]],
+                    [3, [10, 20], [-1, 10, 20], "baf"])
+}
+
+
+def get_all_combinations(rest_arg_names, params):
+    """Recursively get all combinations of arguments."""
+    this_param = rest_arg_names[0]
+    good_values, bad_values = params[this_param]
+    if len(rest_arg_names) == 1:
+        for value in good_values:
+            yield {this_param: value}, True
+        for value in bad_values:
+            yield {this_param: value}, False
+    else:
+        rest_combinations = get_all_combinations(rest_arg_names[1:], params)
+        for combination, good in rest_combinations:
+            for value in good_values:
+                res = {this_param: value}
+                res.update(combination)
+                yield res, good
+            for value in bad_values:
+                res = {this_param: value}
+                res.update(combination)
+                yield res, False
+
+
+class TestEncodersInit(unittest.TestCase):
+    def _construct_all_objects(self, enc_type, params):
+        """Construct all object with given type and param combinations."""
+        arg_names = list(params.keys())
+        print(arg_names)
```

Comment:
Asi by tam nemělo být ani jedno, protože to přetéká maximální velikost logu v Travisu. Na druhou stranu ale potřebujeme vědět, jak vypadala ta konfigurace, na které to spadlo.

Spíš bych se klonil k `print`u, protože ten píše na stdout, zatímco `log` je logování samotné Neural Monkey na stderr.